packages/mermaid/src/dagre-wrapper/markers.js

```diff
@@ -2,6 +2,43 @@
 
 import { log } from '../logger.js';
 
+const getSvgParent = (elem) => {
+  let container = elem;
+
+  // the intent here is to find the first parent element that is NOT part of the SVG element
+  // I know there has to be a better way, but could not find one that worked
+  // tried using checking if elem was instanceof SVGGraphicsElement or SVGElement, but it failed to detect correctly
+  if (container._groups) {
+    container = container._groups[0][0];
+  }
+  if (container.tagName.toLowerCase() === 'g') {
+    container = container.parentElement;
+  }
+  if (container.localName.toLowerCase() === 'svg') {
+    container = container.parentElement;
+  }
+  return container;
```

Comment:
```suggestion
  // the intent here is to find the first parent element that is NOT part of the SVG element
  // I know there has to be a better way, but could not find one that worked
  // tried using checking if elem was instanceof SVGGraphicsElement or SVGElement, but it failed to detect correctly
  if (container._groups) {
    return container._groups[0][0];
  }
  if (container.tagName.toLowerCase() === 'g') {
    return container.parentElement;
  }
  if (container.localName.toLowerCase() === 'svg') {
    return container.parentElement;
  }
  return elem;
```

Reply:
We cant make this change. If container._groups then we need to get the first element in the first element - but that wont get us to the correct element. That just gets us to an element that has a "parentElement" property. So, it still needs the other checks as it goes.  If we are in a "g" element we want the parent, and after THAT, we want the svg element.

I know there is a better way to do this...

Reply:
Ahh, my bad. Those were `if`s, not `else if`s...